.github/workflows/docs.yaml

```diff
@@ -33,21 +30,9 @@ jobs:
       - name: Build documentation
         run: npm run docs:build
 
-  deploy:
-    name: Deploy documentation
-    needs: build
-    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
-    permissions:
-      contents: write
-      pages: write
-      id-token: write
-    runs-on: ubuntu-latest
-    steps:
       - name: Deploy to GitHub Pages
         uses: peaceiris/actions-gh-pages@v4
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           publish_dir: ./docs/.vitepress/dist
-          allow_empty_commit: true
-          keep_files: true
```

Comment:
We should add keep_files back in, and a comment explaining that it's necessary for the coverage reports generated by the testing Action to be preserved by the documentation deployment Action

Reply:
That smells a bit fishy to me.  I attempted to circumvent this problem by having the documentation workflow run after the test workflow has completed (since the coverage reports are needed as part of the documentation).